crates/turbo-malloc/src/counter.rs

```diff
@@ -0,0 +1,72 @@
+use std::{
+    cell::UnsafeCell,
+    ptr::NonNull,
+    sync::atomic::{AtomicUsize, Ordering},
+};
+
+pub static ALLOCATED: AtomicUsize = AtomicUsize::new(0);
+const RESERVE: usize = 1000240; // 100 KiB
+const MAX_RESERVE: usize = 2000480; // 200 KiB
+
+struct ThreadLocalCounter {
+    reserved: usize,
```

Comment:
I would perhaps call this a buffer instead. "Reserved" makes it sound like this counts allocated memory.

```suggestion
    buffer: usize,
```

That or inverse the `add` and `remove` method's logics.